core/trie/proof_test.go

```diff
@@ -1,1139 +1,827 @@
 	})
 
-	t.Run("VP  non existent key - less than root edge", func(t *testing.T) {
-		tempTrie, _ := buildSimpleDoubleBinaryTrie(t)
+	t.Run("left non-existent edge proof", func(t *testing.T) {
+		t.Parallel()
 
-		nonExistentKey := trie.NewKey(123, []byte{0}) // Diverges before the root node (len root node = 249)
-		nonExistentKeyValue := new(felt.Felt).SetUint64(2)
-		proofNodes, err := trie.GetProof(&nonExistentKey, tempTrie)
+		start := 100
+		proof := trie.NewProofNodeSet()
+		err = tr.GetRangeProof(decrementFelt(records[start].key), records[start].key, proof)
 		require.NoError(t, err)
 
-		root, err := tempTrie.Root()
+		_, err = trie.VerifyRangeProof(root, decrementFelt(records[start].key), []*felt.Felt{records[start].key}, []*felt.Felt{records[start].value}, proof)
 		require.NoError(t, err)
-
-		require.False(t, trie.VerifyProof(root, &nonExistentKey, nonExistentKeyValue, proofNodes, crypto.Pedersen))
 	})
 
-	t.Run("VP  non existent leaf key", func(t *testing.T) {
-		tempTrie, _ := buildSimpleDoubleBinaryTrie(t)
+	t.Run("right non-existent edge proof", func(t *testing.T) {
+		t.Parallel()
 
-		nonExistentKeyByte := new(felt.Felt).SetUint64(2).Bytes() // Key not set
-		nonExistentKey := trie.NewKey(251, nonExistentKeyByte[:])
-		nonExistentKeyValue := new(felt.Felt).SetUint64(2)
-		proofNodes, err := trie.GetProof(&nonExistentKey, tempTrie)
+		end := 100
+		proof := trie.NewProofNodeSet()
+		err = tr.GetRangeProof(records[end].key, incrementFelt(records[end].key), proof)
 		require.NoError(t, err)
 
-		root, err := tempTrie.Root()
+		_, err = trie.VerifyRangeProof(root, records[end].key, []*felt.Felt{records[end].key}, []*felt.Felt{records[end].value}, proof)
 		require.NoError(t, err)
-
-		require.False(t, trie.VerifyProof(root, &nonExistentKey, nonExistentKeyValue, proofNodes, crypto.Pedersen))
 	})
-}
 
-func TestProofToPath(t *testing.T) {
-	t.Run("PTP Proof To Path Simple binary trie proof to path", func(t *testing.T) {
-		tempTrie := buildSimpleTrie(t)
-		zeroFeltByte := new(felt.Felt).Bytes()
-		zero := trie.NewKey(250, zeroFeltByte[:])
-		leafValue := utils.HexToFelt(t, "0x0000000000000000000000000000000000000000000000000000000000000002")
-		siblingValue := utils.HexToFelt(t, "0x0000000000000000000000000000000000000000000000000000000000000003")
-		proofNodes := []trie.ProofNode{
-			&trie.Edge{
-				Path:  &zero,
-				Child: utils.HexToFelt(t, "0x05774FA77B3D843AE9167ABD61CF80365A9B2B02218FC2F628494B5BDC9B33B8"),
-			},
-			&trie.Binary{
-				LeftHash:  leafValue,
-				RightHash: siblingValue,
-			},
-		}
+	t.Run("both non-existent edge proofs", func(t *testing.T) {
+		t.Parallel()
 
-		zeroFeltBytes := new(felt.Felt).SetUint64(0).Bytes()
-		leafkey := trie.NewKey(251, zeroFeltBytes[:])
-		sns, err := trie.ProofToPath(proofNodes, &leafkey, crypto.Pedersen)
+		start := 100
+		first, last := decrementFelt(records[start].key), incrementFelt(records[start].key)
```

Comment:
Would this break if by chance first and last correspond to records[start-1].key, records[start+1].key? I guess it's unlikely, but possible? (same for other tests that use decrementFelt etc)

Reply:
Very unlikely. The possibility of that happening is about `(1/2^32)^3`. I used a constant seed for the random test cases so this should be fine.

Reply:
Yup, good point :D